diff --git a/src/builtin.c b/src/builtin.c
index 902490d..a3d41fd 100644
--- a/src/builtin.c
+++ b/src/builtin.c
@@ -1,4 +1,4 @@
-#ifndef __sun__
+#if !defined(__sun__) && !defined(__MVS__)
 # define _XOPEN_SOURCE
 # define _XOPEN_SOURCE_EXTENDED 1
 #else
@@ -1354,13 +1354,18 @@ static time_t my_mktime(struct tm *tm) {
   return _mkgmtime(tm);
 #else
   char *tz;
-
   tz = (tz = getenv("TZ")) != NULL ? strdup(tz) : NULL;
-  if (tz != NULL)
+  
+  if (tz != NULL) {
     setenv("TZ", "", 1);
+  }
+
   time_t t = mktime(tm);
-  if (tz != NULL)
+  
+  if (tz != NULL) {
     setenv("TZ", tz, 1);
+    free(tz);
+  }
   return t;
 #endif
 }
