diff --git a/tests/jq.test b/tests/jq.test
index 4ecf72f..10f5fea 100644
--- a/tests/jq.test
+++ b/tests/jq.test
@@ -1848,9 +1848,10 @@ try ["OK", strflocaltime({})] catch ["KO", .]
 "2015-03-05T23:51:47Z"
 [[2015,2,5,23,51,47,4,63],1425599507]
 
-[strptime("%FT%T")|(.,mktime)]
-"2025-06-07T08:09:10"
-[[2025,5,7,8,9,10,6,157],1749283750]
+# %F conversion specifier not supported on z/OS
+#[strptime("%FT%T")|(.,mktime)]
+#"2025-06-07T08:09:10"
+#[[2025,5,7,8,9,10,6,157],1749283750]
 
 # Check day-of-week and day of year computations
 # (should trip an assert if this fails)
