diff --git a/src/jq_test.c b/src/jq_test.c
index 3945686..db060b2 100644
--- a/src/jq_test.c
+++ b/src/jq_test.c
@@ -318,7 +318,11 @@ static void run_jq_pthread_tests() {
 
     // wait for all threads
     for(a = 0; a < NUMBER_OF_THREADS; ++a) {
+    #ifdef __MVS__
+        if (threads[a].__ != 0) {
+    #else
         if (threads[a] != 0) {
+    #endif
             pthread_join(threads[a], NULL);
         }
     }
