diff --git a/tests/shtest b/tests/shtest
index 14aafbf..98af6c5 100755
--- a/tests/shtest
+++ b/tests/shtest
@@ -1,4 +1,4 @@
-#!/bin/sh -x
+/bin/env $SHELL
 
 . "${0%/*}/setup" "$@"
 
@@ -423,7 +423,9 @@ if [ "$($VALGRIND $Q $JQ -n '{"a":"xyz"} | halt_error(1)' 2>&1)" != '{"a":"xyz"}
 fi
 
 # Check $JQ_COLORS
-unset JQ_COLORS
+if [ -n "${JQ_COLORS+x}" ]; then
+	unset JQ_COLORS
+fi
 
 ## Default colors, null input
 $JQ -Ccn . > $d/color
