diff --git a/scripts/version b/scripts/version
index f24a18d..0b880e0 100755
--- a/scripts/version
+++ b/scripts/version
@@ -5,9 +5,9 @@ cd "$(dirname "$0")/../"
 test -e .git || exit 1
 
 if git describe --tags --match 'jq-*' >/dev/null 2>&1; then
-  git describe --tags --match 'jq-*' --dirty | sed 's/^jq-//'
+  git describe --tags --match 'jq-*' | sed 's/^jq-//'
 else
   branch=$(git rev-parse --abbrev-ref HEAD)
-  commit=$(git describe --always --dirty)
+  commit=$(git describe --always)
   echo "${branch}-${commit}"
 fi
